### DISCUSSION TITLE
e.stopPropagation doesn't work inside nextjs Link component

### DISCUSSION THREAD

**Discussion by zwelhtetyan** [vercel/next.js#45588] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

____

### Link to the code that reproduces this issue

https://github.com/zwelhtetyan/tiktok

### To Reproduce

_____

### Describe the Bug

![Screenshot from 2023-02-02 16-15-26](https://user-images.githubusercontent.com/89287841/216292369-8286a452-a59d-48e7-b133-473c94049ec6.png)

### Expected Behavior

shouldn't go back to index route when click button that stop propagation

### Which browser are you using? (if relevant)

chrome

**Comment by stubbies** [comment, marked as answer]
You need `e.preventDefault()`

**Reply by zwelhtetyan** [reply]
> You need `e.preventDefault()`

It works!!!! Thank you

**Reply by jinyoung4478** [reply]
Great! Why is that?

**Reply by minhnguyenduy99** [reply]
> Great! Why is that?

I cannot confirm but I found this on mozilla documentation:

![image](https://github.com/user-attachments/assets/7ed9de20-0914-4d96-8e43-ec6895090c74)

Even though `stopPropagation` method stops the bubbling, the default behavior/action of `<a>` still occurs, which is to navigate to the link specified in the `href` attribute.

**Comment by LovaArutinovi** [comment]
if previous answer didn't work for someone else, you can also try this
`e.nativeEvent.stopImmediatePropagation();`
